src/prefect/cli/deployment.py

```diff
@@ -566,12 +570,16 @@ async def build(
         )
 
     if not skip_upload:
-        file_count = await deployment.upload_to_storage()
-        if file_count:
-            app.console.print(
-                f"Successfully uploaded {file_count} files to {deployment.location}",
-                style="green",
-            )
+        if (
+            deployment.storage
+            and "put-directory" in deployment.storage.get_block_capabilities()
+        ):
```

Comment:
This side also feels like it needs at least a message saying 'skipping upload since the block is read-only', but it feels wrong to fail the command (possibly more reason to deprecate this flag).

Reply:
updated